conn.go

```diff
@@ -856,65 +948,79 @@ func (c *Conn) readLoop(name string, pktConn ipPacketConn, inboundBufferSize int
 							// Destination: 224.0.0.251
 							// Interface Index: 1
 							// Interface Addresses @ 1: [127.0.0.1/8 ::1/128]
-							if ifIndex != 0 {
+							if ifIndex != -1 {
```

Comment:
@Sean-Der this may have been related to the link-local address issue you mentioned

Reply:
nice! Yea things now pass for me locally